includes/multi-currency/class-multi-currency.php

```diff
@@ -244,9 +289,16 @@ private function initialize_available_currencies() {
 		$woocommerce_currency                                = get_woocommerce_currency();
 		$this->available_currencies[ $woocommerce_currency ] = new Currency( $woocommerce_currency, 1.0 );
 
-		$currencies = $this->get_stored_currencies();
-		foreach ( $currencies as $currency ) {
-			$new_currency                                      = new Currency( $currency[0], $currency[1] );
+		$available_currencies = [];
+
+		$wc_currencies = get_woocommerce_currencies();
+		$cache_data    = $this->get_cached_currencies();
+
+		foreach ( $wc_currencies as $currency_code => $currency_name ) {
+			$currency_rate = $cache_data['currencies'][ $currency_code ] ?? 0.0;
```

Comment:
Yes, for sure. My original thought was 0.0 would make it clear to the merchant that the rate couldn't be fetched and they should set a manual rate (there'd also be some visual hint to this on the frontend), but I think 1.0 is safer to prevent cases where the merchant misses that or accidentally sets up a currency etc.

Updated in bc669d59463005179cf453f138ffdb595bdffee4, thanks for testing!